src/Components/Components/src/Routing/Router.cs

```diff
@@ -215,12 +236,13 @@ internal virtual void Refresh(bool isNavigationIntercepted)
             var routeData = new RouteData(
                 context.Handler,
                 context.Parameters ?? _emptyParametersDictionary);
+
             _renderHandle.Render(Found(routeData));
```

Comment:
Please consider pass the `routeData` into RoutingStateProvider, so that customer can get the routedata in any componenets.

Reply:
@ElderJames that's not the appropriate pattern, if we do something like that, we will expose it like a cascading value, not as a service.